x-pack/plugin/ccr/src/main/java/org/elasticsearch/xpack/ccr/rest/RestCcrStatsAction.java

```diff
@@ -32,7 +38,17 @@ public String getName() {
     @Override
     protected RestChannelConsumer prepareRequest(final RestRequest restRequest, final NodeClient client) {
         final CcrStatsAction.Request request = new CcrStatsAction.Request();
-        return channel -> client.execute(CcrStatsAction.INSTANCE, request, new RestToXContentListener<>(channel));
+        return channel -> client.execute(
+            CcrStatsAction.INSTANCE,
+            request,
+            new ThreadedActionListener<>(
```

Comment:
Why do I still need a threaded listener with chunked encoding?

Reply:
The `toXContent` implementation collects all the shard responses into a map keyed by index up-front - I think that should be done elsewhere.

https://github.com/elastic/elasticsearch/blob/c1efe471fb7d91e0275882bf7339bc2ce04aff66/x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ccr/action/FollowStatsAction.java#L73-L78